### PR TITLE
Add support for cluster install on Alibaba Cloud

### PR DESCRIPTION
**Pull request by akhil-rane**
- [x] Add Alibaba Cloud APIs
- [x] Implement Alibaba clusterresource builder support
- [x] Add controller code to provision/deprovision clusters on Alibaba cloud
- [x] Add hiveutil support for Alibaba cloud

x-ref: https://issues.redhat.com/browse/HIVE-1609
x-ref: https://issues.redhat.com/browse/HIVE-1610
x-ref: https://issues.redhat.com/browse/HIVE-1611

### CODE REVIEW

**Comment by akhil-rane** [comment]
/assign @abutcher 

just includes APIs for now. Working on install code

**Comment by abutcher** [comment]
API changes look good to me :+1:

**Comment by akhil-rane** [comment]
@abutcher This is ready for a review.

**Review by abutcher — COMMENTED** [review]
Looks good, a few small things. Is there anything outstanding?

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ GOFLAGS="" bash ${CODEGEN_PKG}/generate-groups.sh "all" \
 GOFLAGS="" bash ${CODEGEN_PKG}/generate-groups.sh "deepcopy" \
   github.com/openshift/hive/pkg/client \
   github.com/openshift/hive/apis \
-  "hive:v1/agent hive:v1/aws hive:v1/azure hive:v1/baremetal hive:v1/gcp hive:v1/none hive:v1/openstack hive:v1/ovirt hive:v1/vsphere hive:v1/ibmcloud hivecontracts:v1alpha1" \
+  "hive:v1/agent hive:v1/alibabacloud hive:v1/aws hive:v1/azure hive:v1/baremetal hive:v1/gcp hive:v1/openstack hive:v1/ovirt hive:v1/vsphere hive:v1/ibmcloud hivecontracts:v1alpha1" \
```
[the remark this answers is not in the document]
Comment (akhil-rane) [reply hack/update-codegen.sh:22]: I do not know how that got removed but I will add it

**Pull request merged**